cluster/tenants.go

```diff
@@ -35,6 +36,7 @@ type Tenant struct {
 	ID        uuid.UUID
 	Name      string
 	ShortName string
+	Domain    string
```

Comment:
We have ID and ShortName bot being random generated strings, seems like domain is doing the same behavior that shortname used to do (tenant identifier defined by the user), why do we need the domain field now?

Reply:
Probably because of DNS requirements, since they are decoupled it is perhaps possible to use uuid.

Reply:
ID is an identifier, short name will be renamed later, since it's meant to be an identifiable name we can use inside kubernetes that doesn't max out the name length constraints on k8s, domain is what the tenant used to elect before as short name, but since now we use that to track all the internals of a tenant, domain was added to deal with that